ide/src/focus.ts

```diff
@@ -147,8 +154,13 @@ export let focus = async (call_flowistry: CallFlowistry) => {
     clear_ranges(vscode.window.activeTextEditor!);
     state.disposable.dispose();
     state = null;
+
+    render_status_bar(focus_status_bar_item, FocusStatusBarState.Inactive);
+    doc_save_callback.dispose();
   } else {
     await initialize(call_flowistry);
     render();
+
+    doc_save_callback = vscode.workspace.onDidSaveTextDocument(() => initialize(call_flowistry, true));
```

Comment:
This will probably be changed when I work on the UI during/between edits, but for now I just reinitialize focus mode after each document save. The only different between the initialization from user input (win + r, win + a) and initialization from document saves is that the latter won't display compiler output if the program contains errors.

I've also only tested this on simple functions in a small crate so I haven't run into performance issues which will happen when reinitializing after every save.